superset/connectors/sqla/utils.py

```diff
@@ -119,3 +120,22 @@ def get_virtual_table_metadata(dataset: "SqlaTable") -> List[Dict[str, str]]:
     except Exception as ex:
         raise SupersetGenericDBErrorException(message=str(ex)) from ex
     return cols
+
+
+def allow_adhoc_subquery(raw_sql: str) -> bool:
```

Comment:
Ah, I was confused when I saw the use of this function on the file above (`superset/connectors/sqla/models.py`), I expected this to return `True` or `False`, but it never returns `False`.

I think it might be clearer to rename this function and change its signature to something like:

```python
def validate_adhoc_subquery(raw_sql: str) -> None
```

Then in the file above it will be easier to understand what's going on.

Also, can you add a docstring to this function?